### PR TITLE
Fix Windows ISO rename issue

### PR DESCRIPTION
**Pull request by nathanleclaire**
Closes https://github.com/docker/machine/issues/1112.

cc @ehazlett @tianon 

Signed-off-by: Nathan LeClaire <nathan.leclaire@gmail.com>

### CODE REVIEW

**Comment by tianon** [comment]
:+1: logic seems sane to me!

**Comment by nathanleclaire** [comment]
travis 
image

File: utils/b2d.go [hunk 1]
```diff
@@ -132,7 +136,16 @@ func (b *B2dUtils) DownloadISO(dir, file, isoUrl string) error {
...
 	}
 
-	if err := os.Rename(f.Name(), filepath.Join(dir, file)); err != nil {
+	// Dest is the final path of the boot2docker.iso file.
+	dest := filepath.Join(dir, file)
+
+	// Windows can't rename in place, so remove the old file before
+	// renaming the temporary downloaded file.
+	if err := os.Remove(dest); err != nil {
+		return err
```
Comment (tianon) [inline utils/b2d.go]: If the file doesn't exist....  :wink:

Comment (tianon) [inline utils/b2d.go]: `os.IsNotExist` sounds appropriate

**Comment by nathanleclaire** [comment]
Ahahaha I bet this error was getting returned before and it's just that this PR now checks for it

**Comment by nathanleclaire** [comment]
OK Travis go get failed?  I'll try again.

File: utils/b2d.go [hunk 2]
```diff
@@ -94,6 +94,14 @@ func (b *B2dUtils) GetLatestBoot2DockerReleaseURL() (string, error) {
 	return isoUrl, nil
 }
 
+func removeFileIfExists(string name) {
+	if err := os.Stat(name); err == nil {
+		if err := os.Remove(name); err != nil {
```
Comment (tianon) [inline utils/b2d.go]: This is a race...  Why not use [`os.IsNotExist`](https://golang.org/pkg/os/#IsNotExist)?

Comment (tianon) [inline utils/b2d.go]: ie:
```go
if err := os.Remove(name); err != nil && !os.IsNotExist(err) {
    log.Fatalf("Error removing temporary download file: %s", err)
}
```

Comment (nathanleclaire) [inline utils/b2d.go]: Ah, brainfart / overconfidence.  Nice catch

**Comment by nathanleclaire** [comment]
Argh, I'm flailing on this one.  One sec...

**Comment by nathanleclaire** [comment]
OK looks good now! :)

**Comment by ehazlett** [comment]
Tested and works as expected.  LGTM

**Pull request merged**